### PR TITLE
Preliminary work on encryption

### PR DESCRIPTION
**Pull request by jbaublitz**
Supersedes #1817

### CODE REVIEW

File: src/engine/strat_engine/backstore/crypt.rs [hunk 1]
```diff
@@ -329,6 +329,15 @@ impl CryptHandle {
         let name = self.name.to_owned();
         ensure_wiped(&mut self.device, &physical_path, &name)
     }
+
+    pub fn logical_device_size(&mut self) -> Result<u64> {
```
[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/engine/strat_engine/backstore/crypt.rs]: Resolved. I even changed the return type now that I'm confident the units are in sectors and added some additional documentation to other methods in the crypt module.

**Comment by jbaublitz** [comment]
Rebased.

**Comment by jbaublitz** [comment]
Blocked by #1931

**Comment by mulkieran** [comment]
We'll want to merge #1939 and #1932 first.

**Comment by jbaublitz** [comment]
> Please see #1944. The first commit is definite, but the second commit may be somehow insecure, so we need to wait on further information about that.

The key description is not private information as the keyring has proper access controls built into the system. While a keyfile path could be considered insecure as it would lead you to a file that may or may not have proper access controls in place, even if you have the key description, part of the benefit of using the kernel keyring is that it does implement proper access controls so that users outside of whatever session, process, or user added the keyring (depending on how the keyring was configured) will not be able to access the data of the key itself. Unlike with the keyfile path which may or may not be useful, the key description will be useless to attackers.

**Comment by mulkieran** [comment]
> > Please see #1944. The first commit is definite, but the second commit may be somehow insecure, so we need to wait on further information about that.
> 
> The key description is not private information as the keyring has proper access controls built into the system. While a keyfile path could be considered insecure as it would lead you to a file that may or may not have proper access controls in place, even if you have the key description, part of the benefit of using the kernel keyring is that it does implement proper access controls so that users outside of whatever session, process, or user added the keyring (depending on how the keyring was configured) will not be able to access the data of the key itself. Unlike with the keyfile path which may or may not be useful, the key description will be useless to attackers.

I am inclined to believe that that is correct, but I'ld still like to wait for confirmation.

**Comment by jbaublitz** [comment]
> > > Please see #1944. The first commit is definite, but the second commit may be somehow insecure, so we need to wait on further information about that.
> > 
> > 
> > The key description is not private information as the keyring has proper access controls built into the system. While a keyfile path could be considered insecure as it would lead you to a file that may or may not have proper access controls in place, even if you have the key description, part of the benefit of using the kernel keyring is that it does implement proper access controls so that users outside of whatever session, process, or user added the keyring (depending on how the keyring was configured) will not be able to access the data of the key itself. Unlike with the keyfile path which may or may not be useful, the key description will be useless to attackers.
> 
> I am inclined to believe that that is correct, but I'ld still like to wait for confirmation.

Okay sounds good!

**Comment by jbaublitz** [comment]
Rebased.

**Comment by jbaublitz** [comment]
@mulkieran Applied both commits and made one change to one of the commits.

**Pull request closed without merging**